File: src/Managers/UserManager.cpp
#include"UserManager.h"

#include<iostream>
#include<string>
#include<vector>

using std::string;
using std::vector;

using Models::IcarusAction;
using Models::User;

namespace Managers
{
	#pragma
	UserManager::UserManager(User user)
	{
		this->user = user;
	}
	UserManager::UserManager(const IcarusAction icaAct)
	{
		this->icaAction = icaAct;
		this->user = User{};
		parseUserFromActions();
	}
	#pragma Constructors


	#pragma
	User UserManager::retrieveUser() const
	{
		return user;
	}

	void UserManager::parseUserFromActions()
	{
		auto args = icaAction.flags;

		for (auto arg : args)
		{
			auto flag = arg.flag;
			if (flag.compare("-u") == 0)
			{
				user.username = arg.value;
			}
			if (flag.compare("-p") == 0)
			{
				user.password = arg.value;
			}
		}
	}
	#pragma Functions
}

File: src/managers/user_manager.rs
use std::default::Default;

use crate::models::{self};

#[derive(Debug, Default)]
pub struct UserManager {
    pub user: icarus_models::user::User,
    pub ica_action: models::icarus_action::IcarusAction,
}

impl UserManager {
    pub fn retrieve_user(&self) -> icarus_models::user::User {
        self.user.clone()
    }

    pub fn parse_user_from_actions(&mut self) {
        let args = &self.ica_action.flags;

        // Quit the loop when two are found
        let mut amt: i32 = 0;
        for arg in args {
            let flag = &arg.flag;

            if flag == "-u" {
                self.user.username = String::from(&arg.value);
                amt += 1;
            } else if flag == "-p" {
                self.user.password = String::from(&arg.value);
                amt += 1;
            }

            if amt == 2 {
                break;
            }
        }
    }
}
